books_project/books_app/components/author_component.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from books_app.repositories.author_repository import AuthorRepository
+from rest_framework.exceptions import PermissionDenied, NotFound
+
+class AuthorComponent:
+
+    def create_author(self, name , birth_date, nationality):
+        author_birth_date = datetime.strptime(birth_date, '%Y-%m-%d').date()  # Converts string to date object
+        AuthorRepository.create_author(name,author_birth_date,nationality)
+
+    def update_author(self, author_id, name, birth_date, nationality):
+        author = AuthorRepository.get_author(pk=author_id)
+
+        if author is None:
+            raise NotFound("book not found", code=404)
+        AuthorRepository.update_author(id, name, birth_date, nationality)
```

Comment:
There's no id variable here. id is a reserved name in python and we should always avoid using it, instead we use author_id etc..